modules/tinymce/src/plugins/table/main/ts/ui/UiUtils.ts

```diff
@@ -33,23 +30,24 @@ const onSetupToggle = (editor: Editor, formatName: string, formatValue: string)
   };
 };
 
-const filterNoneItem = <T extends ValueItem>(list: T[]) =>
+const filterNoneItem = <T extends Item>(list: T[]) =>
   Arr.filter(list, (item) => Strings.isNotEmpty(item.value));
 
-const generateItem = (editor: Editor, command: string, format: string, item: Item): Menu.ToggleMenuItemSpec => ({
-  text: item.title,
+const generateItem = <T extends Item>(editor: Editor, item: T, format: string, extractText: (item: T) => string, onAction: (item: T) => void): Menu.ToggleMenuItemSpec => ({
+  text: extractText(item),
   type: 'togglemenuitem',
   onAction: () => {
-    editor.execCommand(command, false, item.value);
+    onAction(item);
   },
```

Comment:
This could be reduced to `onAction: () => onAction(item),` or `onAction: Fun.curry(onAction, item)` if you prefer.